neo/SmartContract/ApplicationEngine.cs

```diff
@@ -9,18 +10,17 @@ namespace Neo.SmartContract
     public class ApplicationEngine : ExecutionEngine
     {
         private const long ratio = 100000;
-        private const long gas_free = 10 * 100000000;
+        private static readonly long gas_free = 10 * (long)NativeContract.GAS.Factor;
```

Comment:
all gas should be `ulong`

Reply:
We keep it as `long` for easy subtraction calculations. (Because the subtraction may be negative.)